packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -336,6 +336,52 @@ export const getCartDetails = payload => {
     };
 };
 
+export const retrieveAndMergeCarts = payload =>
+    async function thunk(dispatch) {
+        const { mergeCarts, fetchCartId } = payload;
+        const sourceCartId = await retrieveCartId();
+        try {
+            await dispatch(removeCart());
+            await dispatch(
+                createCart({
+                    fetchCartId
+                })
+            );
+        } catch (error) {
+            dispatch(actions.mergeCarts.receive(error));
+            return thunk(...arguments);
+        }
+
+        try {
+            // errors can come from graphql and are not thrown
+            const destinationCartId = await retrieveCartId();
+            await dispatch(removeCart());
```

Comment:
Why do you remove the cart? Can we just call `mergeCarts` and have the result overwrite the old data? I don't think we need to call the `remove` action here.

Reply:
It didn't rewrite for me at some places, sure I will try removing this.

Reply:
Hey @sirugh can we use this example, to let mutation update cart ids
https://www.apollographql.com/docs/tutorial/mutations/#expose-apollo-client-with-useapolloclient

Reply:
Removed usage of redux.

We still do call `removeCart` in the talon, but only to wipe redux of any stale cart data. Once we call `mergeCarts`, the new user's cart id will have all the info we need, so we then call `getCartDetails` and it updates the redux store. Eventually those calls won't be necessary since we will only be using GQL cache for rendering.